transport.go

```diff
@@ -2567,6 +2568,11 @@ func (pc *persistConn) roundTrip(req *transportRequest) (resp *Response, err err
 		req.extraHeaders().Set("Accept-Encoding", "gzip, deflate, br")
 	}
 
+	// Set gzip to true if the caller's Accept-Encoding includes gzip.
+	if strings.Contains(req.Header.Get("Accept-Encoding"), "gzip") || strings.Contains(req.Header.get("accept-encoding"), "gzip") {
```

Comment:
As the comments of `Header.Get` states, key is case-insensitive as `textproto.CanonicalMIMEHeaderKey` is used to canonicalize the provided key.
So the first check is enough.